### PR TITLE
fix: Enables strict mode for zoom-to-fit

### LINKED ISSUE #2033: Enable strict mode for zoom-to-fit
### Check for duplicates

- [X] I have searched for similar issues before opening a new one.

### Component

zoom-to-fit

### Problem

_No response_

### Request

The zoom-to-fit plugin does not have "[strict mode](https://www.typescriptlang.org/tsconfig#strict)" enabled for TypeScript. We should enable it to have better type checking for our plugins, but doing so may show some TypeScript errors that need to be fixed first.

This would be a good first issue for anyone experienced with working in TypeScript. You may need to adjust types, add checks for null or undefined, or make other fixes. If you have questions, feel free to ask here! In general, the team prefers avoiding casts and using `!` to coerce non-null types, and instead adding appropriate checks.

To work on this issue:

1. Run `npm install` at the root of Blockly samples to set up dependencies.
2. Navigate to the plugin's directory
1. Change `strict: false` to `strict: true` in the plugin's tsconfig.json
2. Run `npm run build` and see if there are any TypeScript errors.
3. Resolve any TypeScript errors, if present. If not, then we got lucky and accidentally wrote strict code! Yay!
4. Run `npm run build` again to make sure there are no TypeScript errors.
5. Run `npm run start` and play with the plugin in the playground to make sure the plugin still behaves as expected.
6. Run `npm run test` to run any automated tests.
7. Run `npm run lint` to make sure there are no formatting errors.
8. Open a PR with your changes!

### Alternatives considered

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by kapooraryan**
## The basics

- [x] I [validated my changes](https://developers.google.com/blockly/guides/contribute/samples#making_and_verifying_a_change)

### Resolves

Closes #2033 

### Proposed Changes

Enables strict mode for the zoom-to-fit plugin by configuring "strict": true in tsconfig.json.

Typescript errors that arose as a result:
Error_TS

Solved by making changes to index.ts in the src and test directory:
Solved_TS

### Reason for Changes

Apart from enabling strict mode, the code now keeps a check for a null object in src/index.ts and a null element in test/index.ts.

### Additional Information

Ran npm run lint and resolved any formatting issues as well. Please do let me know if this is okay or any changes need to be made!

### CODE REVIEW

**Comment by kapooraryan** [comment]
@BeksOmega Ready for review

File: plugins/zoom-to-fit/src/index.ts [hunk 1]
```diff
@@ -198,9 +198,10 @@ export class ZoomToFitControl implements Blockly.IPositionable {
         i = -1;
       }
     }
-
-    this.svgGroup.setAttribute('transform',
-        `translate(${this.left}, ${this.top})`);
+    if (this.svgGroup) {
+      this.svgGroup.setAttribute('transform',
+          `translate(${this.left}, ${this.top})`);
+    }
```
Comment (BeksOmega) [inline plugins/zoom-to-fit/src/index.ts]: Nit: make this a bit  more concise.

```suggestion
      this.svgGroup?.setAttribute('transform',
          `translate(${this.left}, ${this.top})`);
```

**Review by BeksOmega — APPROVED** [review]
This LGTM! Thank you for your work on this :D

This is dependent on https://github.com/google/blockly-samples/pull/2026 so I'm going to hold this for now. You may need to rebase after that gets merged!

**Comment by BeksOmega** [comment]
@kapooraryan I think this needs to be rebased and then should be gtg!

**Comment by kapooraryan** [comment]
Hey, @BeksOmega I have done the rebase and the PR is ready for review. Reference branch: [Branch Link](https://github.com/kapooraryan/blockly-samples/tree/fixes-issue-2033). Edit: [Reference](https://akrabat.com/the-beginners-guide-to-rebasing-your-pr/) I used for the rebase but if anything seems to be wrong, do guide me on how to proceed ahead and thanks.

**Comment by BeksOmega** [comment]
@kapooraryan doesn't look like the rebase fixed the merge conflicts. Can you fix the conflicts? Then I can re-review!

**Comment by kapooraryan** [comment]
> @kapooraryan doesn't look like the rebase fixed the merge conflicts. Can you fix the conflicts? Then I can re-review!

@BeksOmega Done👍.

**Review by BeksOmega — APPROVED** [review]
This LGTM! Thanks for your work on this :D

**Pull request merged**